prow/gerrit/adapter/adapter.go

```diff
@@ -222,44 +223,76 @@ func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, config
 func (c *Controller) Sync() error {
 	syncTime := c.tracker.Current()
 	latest := syncTime.DeepCopy()
+
+	var wg sync.WaitGroup
+	errChan := make(chan error)
+	var repoCacheMapMux, prometheusMux, latestMux sync.Mutex
 	for instance, changes := range c.gc.QueryChanges(syncTime, c.config().Gerrit.RateLimit) {
-		log := logrus.WithField("host", instance)
-		for _, change := range changes {
-			log := log.WithFields(logrus.Fields{
-				"branch":   change.Branch,
-				"change":   change.Number,
-				"repo":     change.Project,
-				"revision": change.CurrentRevision,
-			})
+		wg.Add(1)
+		go func(instance string, changes []gerrit.ChangeInfo) {
+			defer wg.Done()
+			log := logrus.WithField("host", instance)
+			for _, change := range changes {
+				log := log.WithFields(logrus.Fields{
+					"branch":   change.Branch,
+					"change":   change.Number,
+					"repo":     change.Project,
+					"revision": change.CurrentRevision,
+				})
 
-			cloneURI, err := makeCloneURI(instance, change.Project)
-			if err != nil {
-				return fmt.Errorf("makeCloneURI: %w", err)
-			}
+				cloneURI, err := makeCloneURI(instance, change.Project)
+				if err != nil {
+					errChan <- fmt.Errorf("makeCloneURI: %w", err)
+					return
+				}
 
-			cache, ok := c.repoCacheMap[cloneURI.String()]
-			if !ok {
-				if cache, err = createCache(cloneURI, c.cookieFilePath, c.cacheSize, c.configAgent); err != nil {
-					return err
+				repoCacheMapMux.Lock()
+				cache, ok := c.repoCacheMap[cloneURI.String()]
+				if !ok {
+					if cache, err = createCache(cloneURI, c.cookieFilePath, c.cacheSize, c.configAgent); err != nil {
+						repoCacheMapMux.Unlock()
+						errChan <- err
+						return
+					}
+					c.repoCacheMap[cloneURI.String()] = cache
 				}
-				c.repoCacheMap[cloneURI.String()] = cache
-			}
+				repoCacheMapMux.Unlock()
 
-			result := client.ResultSuccess
-			if err := c.processChange(log, instance, change, cloneURI, cache); err != nil {
-				result = client.ResultError
-				log.WithError(err).Info("Failed to process change")
+				result := client.ResultSuccess
+				if err := c.processChange(log, instance, change, cloneURI, cache); err != nil {
```

Comment:
`cache` can be used concurrently safely, right? I think so, but just want to confirm.